nornir/plugins/inventory/netbox.py

```diff
@@ -0,0 +1,80 @@
+import os
+from builtins import super
+
+from nornir.core.inventory import Inventory
+
+import requests
+
+
+class NBInventory(Inventory):
+
+    def __init__(
+        self,
+        nb_url=None,
+        nb_token=None,
+        use_slugs=True,
+        flatten_custom_fields=True,
+        **kwargs
+    ):
+
+        nb_url = nb_url or os.environ.get("NB_URL", "http://localhost:8080")
+        nb_token = nb_token or os.environ.get(
+            "NB_TOKEN", "0123456789abcdef0123456789abcdef01234567"
+        )
+        headers = {"Authorization": "Token {}".format(nb_token)}
+
+        # Create dict of hosts using 'devices' from NetBox
+        devices = {}
+        nb_devices = requests.get(
+            "{}/api/dcim/devices/?limit=0".format(nb_url), headers=headers
+        ).json()
+
+        try:
+            for d in nb_devices["results"]:
+
+                # Create temporary dict
+                temp = {}
+
+                # Add value for IP address
+                try:
+                    temp["nornir_host"] = d["primary_ip"]
+                except TypeError:
```

Comment:
What is the reason this could happen?

Reply:
A `device` can have many `interfaces`, each with many `ip_addresses`.
One `interface` / `ip_address` combination needs to be selected as the `primary_ip` for each `device`.

This is `try`/`except` clause is for when that hasn't happened, then it will return an object with a `None` value for `nornir_host`.

Reply:
Ok, will replace with `temp["nornir_host"] = d.get("primary_ip", None)`